### PR TITLE
added mmap feature (default on). 

### PR DESCRIPTION
**Pull request by s3bk**
Allows to disable memmap dependency when no filesystem is avaible.

(required to build for wasm/asmjs)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -25,9 +29,12 @@ path = "./benches/search.rs"
 test = false
 bench = true
 
+[dependencies.memmap]
+version = "0.4.0"
+optional = true
```
Comment (BurntSushi) [inline Cargo.toml]: Can you write this as `memmap = { version = "0.4.0", optional = true }`? I prefer to keep all deps in one place for easy reading if possible.

File: Cargo.toml [hunk 2]
```diff
@@ -29,12 +29,9 @@ path = "./benches/search.rs"
 test = false
 bench = true
 
-[dependencies.memmap]
-version = "0.4.0"
-optional = true
-
 [dependencies]
 byteorder = "0.5"
+mmap = { version = "0.4.0", optional = true }
```
Comment (BurntSushi) [inline Cargo.toml]: I think the dependency is `memmap`.

Comment (s3bk) [inline Cargo.toml]: yes… I just noticed.

**Comment by BurntSushi** [comment]
Awesome, thanks so much!

**Comment by BurntSushi** [comment]
`fst 0.1.35` is now on crates.io with this PR.

**Pull request merged**